packages/pds/src/db/index.ts

```diff
@@ -1,23 +1,26 @@
 import assert from 'assert'
 import { Kysely, SqliteDialect, PostgresDialect, Migrator } from 'kysely'
 import SqliteDB from 'better-sqlite3'
-import { Pool as PgPool, types as pgTypes } from 'pg'
+import {
+  Pool as PgPool,
+  PoolClient as PgPoolClient,
+  types as pgTypes,
+} from 'pg'
 import DatabaseSchema, { DatabaseSchemaType } from './database-schema'
 import { dummyDialect } from './util'
 import * as migrations from './migrations'
 import { CtxMigrationProvider } from './migrations/provider'
 
 export class Database {
+  listeners: Record<string, ListenerCB[]> = {}
```

Comment:
I wonder if some code could disappear if this were an event-emitter.  Might find that it's easier to type that way too, with [typed-emitter](https://github.com/andywer/typed-emitter).

Reply:
yeah i was feeling this same tension as i started making use of it. gonna take a stab at rewriting w emitters 👌